### PR TITLE
Add providerConfig for `BackupBucket` and `BackupEntry`

### PR DESCRIPTION
**Pull request by rfranzke**
**What this PR does / why we need it**:
This PR adds `providerConfig` and `providerStatus` to the `BackupBucket` and `BackupEntry` resources. This is necessary to allow extension controllers to work with their specific configuration when implementing the `BackupBucket`/`BackupEntry` contract.

**Special notes for your reviewer**:
/cc @mandelsoft @MartinWeindel 

**Release note**:

```noteworthy developer
The `extensions.gardener.cloud/v1alpha1.BackupBucket` resource does now have a new `.spec.providerConfig` field that can be used to pass provider-specific configuration to the extension controller. The `extensions.gardener.cloud/v1alpha1.BackupEntry` resource does now have new `.spec.providerConfig` and `.spec.backupBucketProviderStatus` fields. The contents for both fields are copied over from the respective `BackupBucket` resource. In the garden cluster, the `Seed` resource does now allow to provide the `providerConfig` for backup buckets in `.spec.backup.providerConfig`.
```

### CODE REVIEW

File: docs/extensions/backupentry.md [hunk 1]
```diff
@@ -22,6 +22,10 @@ metadata:
   name: shoot--foo--bar
 spec:
   type: azure
+  providerConfig:
+    <some-optional-provider-specific-backup-bucket-configuration>
+  backupBucketProviderSonfig:
```
Comment (MartinWeindel) [inline docs/extensions/backupentry.md]: typo: backupBucketProviderConfig

**Comment by rfranzke** [comment]
ping for review

File: example/50-seed.yaml [hunk 2]
```diff
@@ -1,43 +0,0 @@
-# Seed cluster registration manifest into which the control planes of Shoot clusters will be deployed.
```
Comment (ialidzhikov) [inline example/50-seed.yaml]: Why we remove the seed example?

Comment (rfranzke) [reply example/50-seed.yaml]: Nice catch, accidentally, will add it again!

Comment (rfranzke) [reply example/50-seed.yaml]: Added again, PTAL.

**Review by ialidzhikov — APPROVED** [review]
/lgtm

**Pull request merged**